### PR TITLE
tests: add a case for invalid context value

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
tests

**Did you add tests for your changes?**
yes
**If relevant, did you update the documentation?**
NA
**Summary**
add a case for invalid context value

**Does this PR introduce a breaking change?**
No

**Other information**
No

### CODE REVIEW

File: test/core-flags/context-flag.test.js [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
 describe('--context flag', () => {
     it('should allow to set context', () => {
-        const { stderr, stdout } = run(__dirname, ['--context', '/test-context-path']);
+        const { stderr, stdout, exitCode } = run(__dirname, ['--context', './']);
 
         expect(stderr).toBeFalsy();
-        expect(stdout).toContain('test-context-path');
+        expect(exitCode).toBe(0);
+        if (isWindows) {
+            expect(stdout).toContain('core-flags');
```
Comment (evilebottnawi) [inline test/core-flags/context-flag.test.js:13]: We can normalize this for windows

Comment (snitin315) [reply test/core-flags/context-flag.test.js:13]: how?

Comment (evilebottnawi) [reply test/core-flags/context-flag.test.js:13]: `string`.replace(/\\/, '/') :smile: Just change windows slashes on linux

File: test/core-flags/context-flag.test.js [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
 describe('--context flag', () => {
     it('should allow to set context', () => {
         const { stderr, stdout, exitCode } = run(__dirname, ['--context', './']);
 
         expect(stderr).toBeFalsy();
         expect(exitCode).toBe(0);
-        if (isWindows) {
-            expect(stdout).toContain('core-flags');
+        if (isWindows && !isWebpack5) {
+            expect(stdout).toContain(`context: 'D:\\\\a\\\\webpack-cli\\\\webpack-cli\\\\test\\\\core-flags'`);
```
Comment (evilebottnawi) [inline test/core-flags/context-flag.test.js:13]:
```
expect(stdout).toContain(`context: '`${path.resolve('./core-flags)}'`)
```

Comment (snitin315) [reply test/core-flags/context-flag.test.js:13]: that works for some windows check but also fails on some windows checks. 😞

Comment (evilebottnawi) [reply test/core-flags/context-flag.test.js:13]: Where is it not work? Having absolute path in tests is really bad idea

Comment (snitin315) [reply test/core-flags/context-flag.test.js:13]: Issue with no of `\`

`{path.resolve('./core-flags)} = D:\\a\\webpack-cli\\webpack-cli\\test\\core-flags'`

https://github.com/webpack/webpack-cli/runs/1346870398?check_suite_focus=true

Comment (evilebottnawi) [reply test/core-flags/context-flag.test.js:13]: Just replace `\` on `\\` after `path.resolve` and it should work

Comment (snitin315) [reply test/core-flags/context-flag.test.js:13]:
```suggestion
            const windowsPath = resolve(__dirname, './').replace(/\\/g, '\\\\\\\\');
            expect(stdout).toContain(windowsPath);
```

File: test/core-flags/context-flag.test.js [hunk 3]
```diff
@@ -1,12 +1,27 @@
...
 describe('--context flag', () => {
     it('should allow to set context', () => {
-        const { stderr, stdout } = run(__dirname, ['--context', '/test-context-path']);
+        const { stderr, stdout, exitCode } = run(__dirname, ['--context', './']);
 
         expect(stderr).toBeFalsy();
-        expect(stdout).toContain('test-context-path');
+        expect(exitCode).toBe(0);
+        if (isWindows) {
+            const windowsPath = resolve(__dirname, './').replace(/\\/g, '\\\\\\\\');
```
Comment (snitin315) [inline test/core-flags/context-flag.test.js:13]:
```suggestion
            const windowsPath = resolve(__dirname, './').replace(/\\/g, '\\\\');
```

**Comment by snitin315** [comment]
fixed for windows.

**Review by evilebottnawi — APPROVED** [review]

**Review by anshumanv — COMMENTED** [review]
Let's do invalid values for all core flags?

**Comment by snitin315** [comment]
Mostly we are using general tests for example for `optimization-*`, `output-*` flags. We use mapping so we actually don't on which flag we are at the moment. 

We have specific test cases only for a few options like `devtool`, `cache`, `records` etc. We can wite invalid values tests in those.

**Comment by anshumanv** [comment]
That makes sense, let's cover others too then

**Comment by evilebottnawi** [comment]
Will we do it in another PR?

**Comment by snitin315** [comment]
Yes, I will check and snatch them all in future PR.

**Comment by anshumanv** [comment]
Sure let's merge this one then

**Review by anshumanv — APPROVED** [review]

**Comment by snitin315** [comment]
Thanks!

**Pull request merged**